src/main/java/teammates/ui/webapi/action/SessionLinksRecoveryAction.java

```diff
@@ -0,0 +1,36 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.output.SessionLinksRecoveryResponseData;
+
+
+/**
+ * Action specifically created for confirming email and sending session recovery links.
+ */
+public class SessionLinksRecoveryAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // no specific access control needed.
+    }
+
+    @Override
+    public ActionResult execute() {
+        String recoveryEmailAddress = getNonNullRequestParamValue(Const.ParamsNames.SESSION_LINKS_RECOVERY_EMAIL);
```

Comment:
my apology for not pointing this out earlier. Should we check whether this is a valid email address?